### ISSUE TITLE
sql: a crash when inserting large rows (probably due to stats collection)

### ISSUE THREAD

**Issue by yuzefovich** [cockroachdb/cockroach#54670]
Using 20.1.1 version, I created a 3 node roachprod cluster with the following schema:
```
create table polling_details (id int primary key, polling_id int, status string, info string);

create index on polling_details(status);

create table polling_actions (id int primary key, polling_detail_id int REFERENCES polling_details(id), status string, info string);

create index on polling_actions(status);
```
Then I kicked off the following script
```
#!/bin/bash

for batch in {1..1000}; do
  ./cockroach sql --insecure -e "INSERT INTO polling_details SELECT i, i % 10, 'IN_PROCESS', repeat('a', 1000000) FROM generate_series($batch*10, $batch*10+9) AS i"
  ./cockroach sql --insecure -e "INSERT INTO polling_actions SELECT i, i, 'COMPLETE', repeat('a', 1000000) FROM generate_series($batch*10, $batch*10+9) AS i"
done
```
which inserts very big rows (I tried first with 5MB rows, and then with 1MB), and both attempts led to a crash of the gateway. I believe it is caused by the automatic stats collection because once I disabled it, I was able to insert the data. This deserves more investigation.

**Comment by yuzefovich** [comment]
I've looked into this, and I think the crashes occur because Go GC is slow in reclaiming the memory under the datums that are used briefly for table statistics collection. At this point, I think the best course of action is replacing existing row-by-row stats collection with a vectorized implementation, and I filed #54803 to do that. Note that simply using `cFetcher` will be of no benefit because we'd still have to convert from `coldata` to `tree.Datum` in order to reuse row-execution processors.
